shared/lib-hoppy/hoppy/async_hoppy_client.py

```diff
@@ -0,0 +1,211 @@
+
+        self.max_latency = max_latency
+        self.response_reject_and_requeue_attempts = response_reject_and_requeue_attempts
+
+        self.async_publisher = AsyncPublisher(config=self.config,
+                                              exchange_properties=exchange_properties,
+                                              queue_properties=request_queue_properties,
+                                              routing_key=request_routing_key)
+
+        self.async_consumer = AsyncConsumer(config=self.config,
+                                            exchange_properties=exchange_properties,
+                                            queue_properties=reply_queue_properties,
+                                            routing_key=reply_routing_key,
+                                            reply_callback=self._on_reply)
+
+        self.responses = {}
+        self.rejected = {}
+
+    def start(self, loop):
+        self.async_publisher.connect(loop)
+        self.async_consumer.connect(loop)
+
+    def stop(self):
+        self.async_publisher.stop()
+        self.async_consumer.stop()
+
+    async def make_request(self, request_id, body):
+        correlation_id = str(uuid.uuid4())
+        logging.info(
+            f"event=requestStarted id={request_id} "
+            f"client={self.name} "
+            f"queue={self.request_queue_properties.name} "
+            f"correlation_id={correlation_id}")
+        self.responses[correlation_id] = None
+
+        self.publish(body, correlation_id, request_id)
+
+        wait_for_response_time = time.time()
+        while True:
+            response = self.responses.get(correlation_id)
+            if response is not None:
+                return self.handle_response(request_id, correlation_id, response)
+            else:
+                self.wait_for_response(correlation_id,
+                                       request_id,
+                                       wait_for_response_time)
+            await asyncio.sleep(0)
+
+    def publish(self, body, correlation_id, request_id):
+        try:
+            self.async_publisher.publish_message(body,
+                                                 BasicProperties(app_id=self.app_id,
+                                                                 content_type="application/json",
+                                                                 reply_to=self.reply_queue_properties.name,
+                                                                 correlation_id=correlation_id))
+        except Exception as e:
+            logging.warning(f"event=requestError "
+                            f"client={self.name} "
+                            f"id={request_id} "
+                            f"queue={self.request_queue_properties.name} "
+                            f"correlation_id={correlation_id} "
+                            f"error='{UNEXPECTED_PUBLISH_ERROR}: {e}'")
```

Comment:
Other than a suggestion to replace `{e}` with `{e!r}`, I'll just leave you with a nagging feeling that there might still be a better way to make these logging statements nicer to read, which would probably involve a bigger overhaul than this PR should have to deal with 😛